---
title: "Module 3"
---

### Bacterial genome assembly data analysis

### Tools to be used in Module 3

**ARIBA**

ARIBA is a tool that identifies antibiotic resistance genes by running local assemblies. It can also be used for MLST calling.

The input is a FASTA file of reference sequences (can be a mix of genes and noncoding sequences) and paired sequencing reads. ARIBA reports which of the reference sequences were found, plus detailed information on the quality of the assemblies and any variants between the sequencing reads and the reference sequences.

Find out more at <https://github.com/sanger-pathogens/ariba/wiki>

#### Multi Locus Sequence Typing (MLST)

-   what is MLST, role and applications

<https://github.com/sanger-pathogens/ariba/wiki/MLST-calling-with-ARIBA>

#### AMR determinants

1.  In your virtual machine, navigate to the folder */home/ubuntu/Data/all_fastas*, where you will find all the assemblies for this study.
2.  Run ARIBA on the samples using the Resfinder database

Eg:

`ariba run ARIBA_dbs/out.resfinder.prepareref all_fastqs/ERR017200_1.fastq.gz all_fastqs/ERR017200_2.fastq.gz ARIBA_output/ERR017200.run`

3.  Write a bash script to run ARIBA on multiple samples at once

4.  Aggregate the ARIBA results
5.  Visualise the results.
6.  Run ARIBA on the samples using the CARD database
7.  Compare Resfinder and CARD results.
