---
title: "Agenda"
---

### Prerequisites for following this course.

1.  You are expected to be comfortable with bash and also with a text editor such as vi or nano. If you are not, try to practice a few commands before you attempt to follow the steps below.

2.  Please make sure you have received from your course instructor the Public IPv4 of your Amazon EC2 instance, the name of the machine you are going to work on and also a public key to access the machine via ssh.

#### Module 1: Set up your bioinformatics working environment 

### Connect to your Amazon EC2 instance

Type the Public IPv4 in your browser address bar, *eg: <https://18.169.193.251/>*

The username is *ubuntu*, and the password the instance ID, *eg i-07833a8ccc08c1030*

Once you log in, you should see the welcome screen.

### How to connect to your Amazon EC2 instance via ssh

Open your Terminal application on your local computer (**not** on the visual interface of your Amazon EC2 Ubuntu machine).

Then type

`ssh -i /local/path/to/key.pem  ubuntu@xxx.xxx.xxx.xxx`

Replace the “xxx.xxx.xxx.xxx” with the Public IPv4 of your Amazon instance and the */local/path/to/* with the local path to your key.pem file

### How to modify your \$PATH variable

Once you are connected to your Amazon EC2 Ubuntu instance, try to view and edit your configuration files. 

Check the configuration files:

`ls -a ~/.`

Edit the ***.profile*** file and add the paths in the **PATH** variable

Eg: 

`vi ~/.profile`

`PATH="/home/ubuntu/Software/mash-Linux64-v2.3:$PATH"`

### Configure your Amazon EC2 Ubuntu instance

#### Install FastQC

#### Install Bactinspector

#### Install the Velvet assembler

Velvet is installed here: 

`/usr/share/miniconda/pkgs/velvet-1.2.10-h5bf99c6_4/bin/velveth`

Add */usr/share/miniconda/pkgs/velvet-1.2.10-h5bf99c6_4/bin/* to **PATH**

#### Install ACT and Artemis

`conda install bioconda::velvet`

#### Install Unicycler 

Manual page: <https://github.com/rrwick/Unicycler?tab=readme-ov-file#build-and-run-without-installation>\

`git clone https://github.com/rrwick/Unicycler.git`

`cd Unicycler`

`python3 setup.py install`

#### Install SPADES

Manual page: <https://github.com/ablab/spades>\

`wget https://github.com/ablab/spades/releases/download/v4.0.0/SPAdes-4.0.0-Linux.tar.gz`

`tar -xzf SPAdes-4.0.0-Linux.tar.gz`

`cd SPAdes-4.0.0-Linux/bin/`

\
Add spades.py to \~/.profile 

####  Install Quast

`conda install -c bioconda quast`

\
Eg on how to run Quast

`quast.py test_data/contigs_1.fasta \`

`test_data/contigs_2.fasta \`

`-r test_data/reference.fasta.gz \`

`-g test_data/genes.txt \`

`-1 test_data/reads1.fastq.gz -2 test_data/reads2.fastq.gz \`

`-o quast_test_output`

Quast path: */home/ubuntu/Software/quast/build/scripts-3.8/quast.py* 

#### Install brew

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

\
`echo >> /home/ubuntu/.zshrc`

`echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' >> /home/ubuntu/.zshrc`

`eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"`\

Install makeblastdb and tblastn

#### Install prokka

\
`brew install brewsci/bio/prokka`

#### Install resfinder

`pip install resfinder`

\
`python /home/ubuntu/.local/lib/python3.9/site-packages/resfinder/run_resfinder.py`

`/usr/share/doc/ncbi-blast+`
