---
title: "Module 2"
---

### Module 2: Data retrieval, Read QC, Genome Assemblies and Assemblies QC

#### Connect to your Amazon EC2 instance

Follow the steps in Module 1 to connect to your remote virtual machine.

The next exercise is broadly based on the study [https://journals.asm.org/doi/full/10.1128/msphere.00185-23](https://journals.asm.org/doi/full/10.1128/msphere.00185-23https://journals.asm.org/doi/full/10.1128/msphere.00185-23) by Abrudan and Shamanna, 2023

#### Data retrieval from the European Nucleotide Archive (ENA)

The European Nucleotide Archive (ENA) provides a comprehensive record of the world’s nucleotide sequencing information, covering raw sequencing data, sequence assembly information and functional annotation.

Access to ENA data is provided through the browser, through search tools, through large scale file download and through the API.

### Download the data programatically

1.  Connect to your Amazon EC2 Ubuntu instance via ssh.

    How to retrieve data for the samples G18255819 and G18252308 from ENA

2.  Create a directory called data in /home/ubuntu/

`mkdir /home/ubuntu/`

3.  Create subdirectories for each of your samples

    `mkdir /home/ubuntu/data/G18255819/illumina`

    `mkdir /home/ubuntu/data/G18255819/nanopore`

    `mkdir /home/ubuntu/data/G18252308/illumina`

    `mkdir /home/ubuntu/data/G18252308/nanopore`\

4.   Download the reads for each sample

`cd /home/ubuntu/data/G18255819/nanopore`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR818/004/ERR8187234/ERR8187234.fastq.gz`

`cd /home/ubuntu/data/G18255819/illumina`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR478/004/ERR4784794/ERR4784794_1.fastq.gz`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR478/004/ERR4784794/ERR4784794_2.fastq.gz`\

`cd /home/ubuntu/data/G18252308/nanopore`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR818/005/ERR8187245/ERR8187245.fastq.gz`

`cd /home/ubuntu/data/G18252308/illumina`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR463/006/ERR4635696/ERR4635696_1.fastq.gz`

`wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR463/006/ERR4635696/ERR4635696_2.fastq.gz`

### Download data through the browser

1.  Connect to your virtual machine via the browser of your host machine.
2.  In your virtual machine, open a browser.
3.  Navigate to <https://www.ebi.ac.uk/ena/browser/home>
4.  Download files associated with the samples G18255819 and G18252308. Downaload the Nanopore and the Illumina reads.

## Reads QC

In this part of the exercise, we will use a programme called [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/).

FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines. It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.

The main functions of FastQC are

-   Import of data from BAM, SAM or FastQ files (any variant)

-   Providing a quick overview to tell you in which areas there may be problems

-   Summary graphs and tables to quickly assess your data

-   Export of results to an HTML based permanent report

### Run FastQC programatically

To run non-interactively you simply have to specify a list of files to process on the command line. 

`fastqc somefile.txt someotherfile.txt`

`fastqc /home/ubuntu/data/*/*/*`

### QC your fastq reads through the visual interface

Navigate to */home/ubuntu/Software/FastQC* and double click on the FastQC icon.

In the visual interface, open all files produced by FastQC and assess the results, eg ERR4635696_1_fastqc.html

Visualize and interpret the FastQC results. Compare the QC results for Nanopore and Illumina sequence reads

Optional: Unzip the files */home/ubuntu/Data/\*/\*/\*\_\*\_fastqc.zip* and produce a script that extracts the information on reads lengths, GC contents etc; plot the results from all samples.

## Speciation

[Bactinspector](https://gitlab.com/antunderwood/bactinspector ) is a package to a) determine the most probable species based on sequence in fasta/fastq files using refseq and Mash (<https://mash.readthedocs.io/en/latest/index.html>) and b) determine the closest reference in refseq to a set of fasta/fastq files.

1.  Go to sample files /*home/ubuntu/Data/* and run Bactinspector using the following command

`bactinspector closest_match -fq "*_2.fastq.gz"`

2.  Check the output \*.tsv file

`tail -n +2 *.tsv| column -t | less -S`\
`cat *.tsv | tr "\t" "~" | cut -d"~" -f2`

*Eg of a result:*

*ASM289538v1*

*GCF_002895385*\

3.  How do you interpret the results? Look up your results in ENA.

## Genome assembly

**The Velvet assembler**

**Velvet** is an algorithm package that has been designed to deal with *de novo* [genome assembly](https://en.wikipedia.org/wiki/Genome_assembly "Genome assembly") and short read [sequencing alignments](https://en.wikipedia.org/wiki/Sequence_alignment "Sequence alignment"). This is achieved through the manipulation of [de Bruijn graphs](https://en.wikipedia.org/wiki/De_Bruijn_graph "De Bruijn graph") for genomic sequence assembly via the removal of errors and the simplification of repeated regions.

1.  Run Velvet

**Visualise the assembles using ACT and Artemis**

ACT is a Java application for displaying pairwise comparisons between two or more DNA sequences.

ACT can be used to identify and analyse regions of similarity and difference between genomes and to explore conservation of synteny, in the context of the entire sequences and their annotation. It can read complete EMBL, GENBANK and GFF entries or sequences in FASTA or raw format.

*Tip: ACT and Artemis are installed here: /usr/share/miniconda/pkgs/artemis-18.2.0-hdfd78af_0/share/artemis-18.2.0-0* 

## Assembly QC with Quast

[QUAST](https://github.com/ablab/quast) stands for QUality ASsessment Tool. It evaluates genome/metagenome assemblies by computing various metrics. The current QUAST toolkit includes the general QUAST tool for genome assemblies, MetaQUAST, the extension for metagenomic datasets, QUAST-LG, the extension for large genomes (e.g., mammalians), and Icarus, the interactive visualizer for these tools.

The QUAST package works both with and without reference genomes. However, it is much more informative if at least a close reference genome is provided along with the assemblies. The tool accepts multiple assemblies, thus is suitable for comparison.
