SARS-CoV-2 surveillance in Romania

Genomic surveillance of pathogens aims to understand the emergence and dissemination of pathogens or their lineages of risk with the ultimate goal of implementing evidence-based interventions to protect public health. Epidemiological data from patients is collected by healthcare professionals. Data on species identification and any phenotypic or molecular characterization of the isolates is often generated by the microbiology laboratories linked to healthcare facilities and/or the reference laboratory. In an ideal scenario, the different sources of laboratory data are stored in a centralised surveillance system and database (such as [WHONET](https://whonet.org/)). However, these systems rarely also incorporate genomic data produced by bioinformaticians. 

Conversely, bioinformaticians often use different analytic tools to extract useful data from the genomes, such as presence/absence of markers of risk (like AMR and virulence genes, and genotyping information (such as MLST, cgMLST, and other species-specific typing), and phylogenetic relationships (trees), as well as quality metrics. 

The genomic epidemiologist’s job often starts by combining data from diverse sources that might not be complete or standardised to facilitate the identification of relevant patterns for meaningful interpretation and intervention. The data and any patterns often need to be digested and summarised for decision makers.

The Centre for Genomic Pathogen Surveillance develops free web applications for integration and visualisation of surveillance (and other) data, called [data-flo](https://data-flo.io/) and [Microreact](http://www.microreact.org). The aim of this module is to highlight the role of the various analytics and sources of data in pathogen surveillance, and how streamlined data integration is essential for real-time decision making, while also introducing resources that can facilitate the inclusion of this topic in teaching curricula.

\

# Learning Outcomes

At the end of this module participants will be able to:

1.  Distinguish the different components of genomic surveillance and their sources

2.  Identify analytic tools to extract from genomes useful information for surveillance 

3.  Utilise data-flo to join different sources of data

4.  Utilise Microreact to recognize the interplay between different types of data

5.  Reflect on the data integration process and its importance for decision making

6.  Recognize activities that highlight the value of data integration for genomic surveillance

\
\

# Case Study: SARS-CoV-2 surveillance in Romania during the first months of the pandemic

\

Note: This case-study is based on real sequence data from  "**Molecular epidemiology analysis of SARS-CoV-2 strains circulating in Romania during the first months of the pandemic**." by Surleac, Marius, et al., <https://doi.org/10.3390/life10080152> , but it has been modified for the purposes of teaching this course. 

\

## **Activities outline**

\

## **Input files:** 

<https://drive.google.com/drive/folders/1yP-ectAQ0plj8G1rcd_UNqPOyFIC7axa?usp=sharing>

\

## **Activity 1. Explore the epidemiological data**

\

24 SARS-CoV-2 samples collected from Romania during the first months of the pandemic have been selected for sequencing. 

\

Explore the epidemiological data. What are your observations?

\

|         |           |                |            |
|---------|-----------|----------------|------------|
| Country | Town      | NAME           | Date       |
| Romania | Bucuresti | EPI_ISL_468134 | 21/03/2020 |
| Romania | Bucuresti | EPI_ISL_468135 | 22/03/2020 |
| Romania | Bucuresti | EPI_ISL_468136 | 22/03/2020 |
| Romania | Suceava   | EPI_ISL_468137 | 22/03/2020 |
| Romania | Suceava   | EPI_ISL_468138 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468139 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468140 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468141 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468142 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468143 | 25/03/2020 |
| Romania | Suceava   | EPI_ISL_468144 | 25/03/2020 |
| Romania | Bucuresti | EPI_ISL_468145 | 28/03/2020 |
| Romania | Suceava   | EPI_ISL_468146 | 08/04/2020 |
| Romania | Bucuresti | EPI_ISL_468147 | 11/04/2020 |
| Romania | Bucuresti | EPI_ISL_468148 | 11/04/2020 |
| Romania | Bucuresti | EPI_ISL_468149 | 12/04/2020 |
| Romania | Bucuresti | EPI_ISL_468150 | 12/04/2020 |
| Romania | Bucuresti | EPI_ISL_468151 | 15/04/2020 |
| Romania | Bucuresti | EPI_ISL_468152 | 16/04/2020 |
| Romania | Bucuresti | EPI_ISL_468153 | 17/04/2020 |
| Romania | Bucuresti | EPI_ISL_468154 | 17/04/2020 |
| Romania | Bucuresti | EPI_ISL_468155 | 18/04/2020 |
| Romania | Suceava   | EPI_ISL_468157 | 29/05/2020 |
| Romania | Suceava   | EPI_ISL_468158 | 29/05/2020 |

\
\

## **Activity 2. Explore the bioinformatics data in Pathogenwatch**

\

NB: If you are not able to upload the sequences to Pathogenwatch and follow steps 2.1-2.4, then you can use a pre-built collection and explore the data presented there: <https://pathogen.watch/collection/5hpqvjar7gfa-sars-cov-2-romania-v1>

\
\

2.1 Download the sequencing data file called **sars-cov-2 romania.fasta** from the shared drive.

\

2.2 Upload the fasta file to <https://pathogen.watch/> .

\

2.3 Select all genomes and create a collection. 

\

2.4 Explore the QC statistics: genome length, N50, number of contigs. 

What do these information mean? Can you trust this data?

\

2.5 Go to the typing tab. What SARS-CoV-2 lineages are present in this collection? What does this tell you?

\

2.6 Go the “Notable mutations” tab. Which “notable mutations” are present in this collection? What does this tell you?

\

## **Activity 3. Link epidemiological data to the bioinformatics analysis**

\

3.1 Pathogenwatch allows you to download the results of the analysis, in the form of four files: 

\

sarscov2Variants.csv

stats.csv

Tree.nwk

typing.csv

\

3.2 In order to analyse the information, you need to aggregate the input files. For this, you can run a pre-made Data-flo <https://data-flo.io/edit?4LxncBtNr69TmfzTGj9Szm> and use the files listed above, together with the metadata file as an input. 

\

**How to run your data-flo**

\

3.3  A step-by-step guide on how to build a data-flo similar to the one above. 

….

\

## **Activity 4. Interactive visualisation with Microreact**

\

For this activity, you can use the Microreact output of the Data-flo from Activity 3, or you can try a ready-made Microreact: <https://microreact.org/project/sars-cov-2-romania> 

\

4.1 Explore the timeline and the map. Where and when was the first case reported?

4.2 Explore the aggregated metadata. 

4.3 Explore the views

4.4 Create a chart

\

## **Activity 5. Interpret the data in a wider context**

\
